core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -52,9 +55,13 @@ object AclCommand {
         listAcl(opts)
     } catch {
       case e: Throwable =>
-        println(s"Error while executing ACL command: ${e.getMessage}")
-        println(Utils.stackTrace(e))
-        System.exit(-1)
+        if (inTestMode) {
```

Comment:
I would much prefer if we would avoid this by calling addACL(opts) directly rather than main.
In general, both the extra switch for testing and unit-tests calling main() method are not recommended if they can at all be avoided, and I think in this case you can easily avoid it.